File: backend/components/metiers/threed/api/samples/hooks/sample_hooks.js
/* global console */
// Exemple ultra avancé clé en main des hooks API Threed (JS)
const { beforeAction, afterAction } = require('../hooks/hooks');
const { auditEntity } = require('../audit/audit');

async function sampleHooksUltra() {
  beforeAction('read', { id: 1 });
  console.info('Avant lecture');
  // ... logique métier ...
  auditEntity({ id: 1 }, 'read');
  afterAction('read', { id: 1 });
  console.info('Après lecture');
  console.log('Hooks ultra avancé exécuté avec succès.');
}

module.exports = { sampleHooksUltra };

File: backend/components/metiers/threed/api/samples/hooks/sample_hooks.py
# Exemple ultra avancé clé en main des hooks API Threed (Python)
from ..hooks.hooks import before_action, after_action
from ..audit.audit import audit_entity
import logging

def sample_hooks_ultra():
    # Hook avant action
    before_action('read', {'id': 1})
    logging.info('Avant lecture')
    # ... logique métier ...
    audit_entity({'id': 1}, 'read')
    # Hook après action
    after_action('read', {'id': 1})
    logging.info('Après lecture')
    print('Hooks ultra avancé exécuté avec succès.')
